### PR TITLE
[libhsplasma] Update to 2024-03-07

### PR DESCRIPTION
**Pull request by zrax**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [X] The "supports" clause reflects platforms that may be fixed by this new version.
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libhsplasma/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
```
Comment (WangWeiLin-MV) [inline ports/libhsplasma/portfile.cmake:1]:
```suggestion
string(REGEX REPLACE "-" "." REF_DOT_VERSION_DATE ${VERSION})

vcpkg_from_github(
```

File: ports/libhsplasma/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO H-uru/libhsplasma
-    REF 448ad712015c0b8a293af7bd56ab623dd2e6e131 # 2023.05.08
-    SHA512 c08e7708f1a6e392075cc6eccbbf2ac932ce77d789ecb192e6805e478fe32f412c4b964494bdcc8574ed18c47b16c02e99b98994a63478ea6768ccedea6a3417
+    REF e043a0e996f709883374d9d479953ffeca37888e # 2024.03.07
+    SHA512 38273a673ddc92d9c82b5476aaaf479e55cfef27a5523c3958cd01e90f8ec20d2f96d714b9aa6c45e00e845ae127d2bdf7f19f499c249547370951977d62fead
```
Comment (WangWeiLin-MV) [inline ports/libhsplasma/portfile.cmake:5]:
```suggestion
    REF "${REF_DOT_VERSION_DATE}"
    SHA512 2edf124fe583e053c078f58d94110ed2285e1f02a34cc7607dfc79f8ab587e173e5782af5d4e2846613d7d6b3e1a27a319fdf138c7546c1c6257b5c8422c2f5a
```

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]

**Review by WangWeiLin-MV — APPROVED** [review]
Port usage and features test pass with following triplets:
* x64-windows

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**